scene/main/viewport.cpp

```diff
@@ -2507,6 +2513,8 @@ void Viewport::_gui_update_mouse_over() {
 		return;
 	}
 
+	gui.sending_mouse_enter_exit_notifications = true;
```

Comment:
In the implementation you keep track in every Viewport separately, if `sending_mouse_enter_exit_notifications` is active or not.
Have you considered the idea to keep track of the state in a global manner (for example only in the root-Viewport)? I'm not entirely sure, if there can be edge cases, where these notifications in different Viewports could interfere with each other.

However, for now it looks like a reasonable approach.